### PR TITLE
Factor out ClientToSVGTranslator.

### PR DESCRIPTION
**Pull request by jtlan**
Factored the position-translation logic out into a separate class,
so it can be reused.

### CODE REVIEW

File: src/utils/clientToSVGTranslator.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      if (translator == null) {
+        translator = new ClientToSVGTranslator(svg);
+        (<any> svg)[ClientToSVGTranslator._TRANSLATOR_KEY] = translator;
+      }
+      return translator;
+    }
+
+    constructor(svg: SVGElement) {
+      this._svg = svg;
+      this._measureRect = <SVGElement> <any>document.createElementNS(svg.namespaceURI, "rect");
```
Comment (bluong) [inline src/utils/clientToSVGTranslator.ts]: Curious, what's with the multiple casting here

File: src/utils/clientToSVGTranslator.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      this._measureRect.setAttribute("class", "measure-rect");
+      this._measureRect.setAttribute("style", "opacity: 0; visibility: hidden;");
+      this._measureRect.setAttribute("width", "1");
+      this._measureRect.setAttribute("height", "1");
+      this._svg.appendChild(this._measureRect);
```
Comment (bluong) [inline src/utils/clientToSVGTranslator.ts]: Curious how much cleaner would this be with using d3?

**Comment by bluong** [comment]
Minor

File: src/utils/clientToSVGTranslator.ts (hunk 1 above)
Comment (jtlan) [inline src/utils/clientToSVGTranslator.ts]: `createElementNS()` supposedly returns an `Element`, even though in this case it returns an `SVGElement`. Can't cast directly from `Element` to `SVGElement`.

File: src/utils/clientToSVGTranslator.ts (hunk 2 above)
Comment (jtlan) [inline src/utils/clientToSVGTranslator.ts]: Possibly somewhat, but I wanted to make this work independently of D3.

**Comment by jtlan** [comment]
Purely a regression pass here; visible effects would be on `Interaction.Key` and `Interaction.Hover`.

**Pull request merged**